internal/controller/children.go

```diff
@@ -64,6 +64,12 @@ func (c ChildResource[T]) Reconcile(ctx context.Context, clnt client.Client, cr
 				err := ctrl.SetControllerReference(cr, current, clnt.Scheme())
 				return err
 			}
+			for i := range desired.Spec.Template.Spec.InitContainers {
+				cont := &desired.Spec.Template.Spec.InitContainers[i]
+				if cont.Name == "" {
```

Comment:
Shouldn't that trigger an error ?

Reply:
This is a leftover from development that I should have removed. I will remove it.